sympy/physics/mechanics/kane.py

```diff
@@ -15,48 +15,87 @@
+
+def _none_handler(x):
+    """Returns and empty matrix if x is False."""
+    if x:
+        return Matrix(x)
+    else:
+        return Matrix()
 
 
 class KanesMethod(object):
-    """Kane's method object.
+    """This class is used automatically form the equations of motion of a
+    multibody system using Kane's method [Kane1985]_.
+
+    The dynamic equations of motion are provided in Kane's form::
+
+        fr + frstar = 0
+
+    in the factored implicit form::
 
-    This object is used to do the "book-keeping" as you go through and form
-    equations of motion in the way Kane presents in:
-    Kane, T., Levinson, D. Dynamics Theory and Applications. 1985 McGraw-Hill
+        mass_matrix * accelerations = forcing
 
-    The dynamical equations of motion in the form::
+    or in the explicit form::
 
-        mass_matrix * speeds.diff() = forcing
+        accelerations = mass_matrix.inv() * forcing
 
-    where the mass matrix, M, and the right hand side vector, f, are available
-    as class attributes after Kane's equations are formed.
+    where fr, frstar, the mass matrix, and the right hand side vector are
+    available as class attributes after Kane's equations are formed.
 
     Attributes
     ==========
 
+    inertial_frame : ReferenceFrame
+        The inertial reference frame the system is defined with respect to.
+    n, num_coordinates, num_speeds : integer
+        The total number of generalized coordinates and speeds.
     q, coordinates : Matrix, shape(n, 1)
-        Matrices of the generalized coordinates.
+        The generalized coordinates. These are ordered as [q_s, q_r]^T.
+    o, num_dependent_coordinates : integer
+        The number of dependent coordinates.
+    q_s, independent_coodrinates : Matrix, shape(n-o, 1)
+        The independent generalized coordinates.
+    q_r, dependent_coodrinates : Matrix, shape(o, 1)
+        The dependent generalized coordinates.
     u, speeds : Matrix, shape(n, 1)
-        Matrices of the generalized speeds. For a non-holonomic system these
-        are ordered as [u_s, u_r].
-    bodies, bodylist : iterable
-        Iterable of Particle and RigidBody objects in the system.
-    loads, forcelist : iterable
-        Iterable of (Point, vector) or (ReferenceFrame, vector) tuples
-        describing the forces and torques on the system.
+        The generalized speeds. For a nonholomic system these are ordered as
+        [u_s, u_r]^T.
+    p, num_indepenent_speeds : integer
+        The number of independen speeds.
+    u_s, independent_speeds : Matrix, shape(p, 1)
+        The independent generalized speeds.
+    m, num_dependent_speeds : integer
+        The number of dependent speeds.
+    u_r, dependent_speeds : Matrix, shape(m, 1)
+        The dependent generalized speeds.
+    holonomic : boolean
+        True if the system is a holonomic system.
+    nonholonomic : boolean
+        True if the system is a nonholonomic system.
```

Comment:
Isn't a system holonomic if it is not nonholonomic? Thus if they are both included there are two different ways to access the same information? I.e. `if holonomic:` is the same as `if not holonomic`

I see one of the advantages of python over other programming languages is it's attempt at a "one method of doing things" for simplicity sake. Otherwise there are too many variations to achieve the same task and at some point it becomes simpler to rewrite code instead of reusing/editing code (it's my understanding this is the state that the Perl language is in). For the same reason I think if we change from u -> speeds, q -> coordinates etc we should deprecate the old attributes. Long run it would reduce clutter and complexity.